File: stock/stock.js
const findMax = (prices) => {
  let profit = 0
  let min    = prices[0]
   
  prices.forEach((price) => {
    if(price < min){
      min = price
    }

    let posProfit = price - min
    if(posProfit > profit){
      profit = posProfit
    }
  })
  return profit
}

let prices = [7,1,5,3,6,4]
console.log(findMax(prices)) // 5

prices = [7,6,4,3,1]
console.log(findMax(prices)) // 0

File: stock/stock.rb
def stock_picker(prices)
  profit = 0
  min    = prices[0]

  prices.each do |price|
    min = price if price < min
    pos_profit = price - min
    profit = pos_profit if pos_profit > profit
  end
  profit
end

prices = [7,1,5,3,6,4] #5

p stock_picker(prices)

prices = [7,6,4,3,1] #0

p stock_picker(prices)
